cmplibrary/src/main/java/com/sourcepoint/cmplibrary/exposed/gpp/GppConfig.kt

```diff
@@ -0,0 +1,45 @@
+package com.sourcepoint.cmplibrary.exposed.gpp
+
+/**
+ * Class that is responsible for storing and passing user's config for GPP legislation
+ */
+data class GppConfig(
+    val coveredTransaction: GppOptionBinary = GppOptionBinary.NO,
+    val optOutOptionMode: GppOptionTernary = GppOptionTernary.NOT_APPLICABLE,
+    val serviceProviderMode: GppOptionTernary = GppOptionTernary.NOT_APPLICABLE,
+) {
+
+    private constructor(
+        builder: Builder
+    ) : this(
+        coveredTransaction = builder.coveredTransaction ?: GppOptionBinary.NO,
+        optOutOptionMode = builder.optOutOptionMode ?: GppOptionTernary.NOT_APPLICABLE,
+        serviceProviderMode = builder.serviceProviderMode ?: GppOptionTernary.NOT_APPLICABLE,
+    )
+
+    class Builder {
```

Comment:
What's the advantage of having a builder for a data class with only 3 attributes?

Reply:
@andresilveirah I created the `Builder` for the `GppConfig` for several reasons:
1. I know it is a good rule of thumb to use creational patterns in order to allow the users of our API's (in our case - our library) to create objects from our API (I am a user of multiple Android libraries, and whenever the library has such a mechanism it gives me a good UX)
2. It provides an option to use this `Builder`, but it is not mandatory, so our users can choose between 2
3. It provides the user an ability to create `GppConfig` with only a couple of parameters (the same with the constructor, but again UX matter)

To summarize, the main goal was to provide the user with more conventional way to instatiate objects from the external API, without making it **mandatory**.

Reply:
I think the added complexity is not worth the benefit. We don't need 2 ways to instantiate the same simple object. The constructor with default values for each attribute is sufficient for ours (and our clients) needs.

The following api is just what we need:
```kotlin
data class GppConfig(
    val a: String? = null,
    val b: String? = null,
    val c: String? = null
)

val config = GppConfig(a: "value")
```

The user can set any value when instantiating the config or leave them untouched.